### PR TITLE
build on all minors

### PR DESCRIPTION
**Pull request by mapsam**
This PR does a few things:

* Updates scripts/generate.sh to grab the package.json version from mapnik rather than hard-coding the version - this means that we'll always be building the minor version from within master. The next time we release a new minor or higher (>= 3.7.x) we'll automatically create a new directory in our documentation with that name
* adds `npm run docs` and `npm run deploy` for easier usage
* automatically populates the node-mapnik/index.html file with the most recent version
* adds 3.4.x documentation which previously didn't exist

cc @springmeyer @artemp @flippmoke @gretacb

### CODE REVIEW

File: scripts/generate.sh [hunk 1]
```diff
@@ -1,15 +1,24 @@
 # generates Node Mapnik documentation using documentationjs
```
Comment (springmeyer) [inline scripts/generate.sh:1]: @mapsam  - I noticed this script does not have `set -eu` at the top- can it (to ensure it stops on error)?

File: node-mapnik/3.6/assets/site.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      element.classList.add('hide');
+    }
+  }
+
+  // var githubAnchors = document.getElementsByClassName('github');
+  // for (var i = 0; i < githubAnchors.length; i++) {
+  //   var elem = githubAnchors[i];
+  //   var url = elem.href;
+  //   url = elem.href.replace('/documentation/', '/node-mapnik/').replace('/node_modules/mapnik', '')
+  // }
```
Comment (springmeyer) [inline node-mapnik/3.6/assets/site.js:46]: Can this code be removed?

**Comment by springmeyer** [comment]
Which, of any, of the existing issues at https://github.com/mapnik/documentation/issues can be closed now?

File: scripts/generate.sh (hunk 1 above)
Comment (mapsam) [reply scripts/generate.sh:1]: Will do!

File: node-mapnik/3.6/assets/site.js (hunk 2 above)
Comment (mapsam) [reply node-mapnik/3.6/assets/site.js:46]: Hmmm looks like this was an old attempt at #6 - I'll work on resolve this now

Comment (mapsam) [reply node-mapnik/3.6/assets/site.js:46]: Fixed in #11

File: scripts/generate.sh (hunk 1 above)
Comment (mapsam) [reply scripts/generate.sh:1]: Fixed in #11

**Pull request merged**